app/services/klarna_service.rb

```diff
@@ -0,0 +1,165 @@
+# Interface to Klarna Checkout service API
+# https://developers.klarna.com/api/#checkout-api
+class KlarnaService
+  require 'httparty'
+
+  SUCCESS_CODES = [200, 201, 202, 204].freeze
+
+  def self.create_order(payment_data)
+
+    raise 'Invalid payment type' unless
+      payment_data[:type] == Payment::PAYMENT_TYPE_MEMBER ||
+      payment_data[:type] == Payment::PAYMENT_TYPE_BRANDING
+
+    response = HTTParty.post(KLARNA_CHECKOUT_URL,
+                             basic_auth: auth,
+                             headers: { 'Content-Type' => 'application/json' },
+                             body: order_json(payment_data))
+
+    return response.parsed_response if SUCCESS_CODES.include?(response.code)
+
+    # Wrap cause exception within HTTP error exception so both appear in log
+    begin
+      process_api_error(response)
+    rescue RuntimeError
```

Comment:
Should try to capture specific errors and then default to a generic StandardError if no others were matched.